### PR TITLE
Add local to global trandform to BinUtility

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
The material in the material map (both surface and volume) should be stored in the local coordinate frame to be able to properly bin the surface/volume if it is not centred on (0,0,0) for cylinders or (0,0,Z) for disks. 

For this the BinUtility used by the map need to have access to the local->global coordinate transform. The Json writer has been modify to store this transformation if it differ from the identity. Such transform is initialise in the Mapper when the BinUtility of the ProtoMaterial is adjusted to the corresponding surface using the transform stored in the surface.

For the Volume map this transform is added into the transform that already exist to go from the global coordinate to the grid coordinate.

### CODE REVIEW

File: Core/include/Acts/Utilities/BinAdjustment.hpp [hunk 1]
```diff
@@ -27,9 +27,13 @@ namespace Acts {
...
 ///
 /// @return new updated BinUtiltiy
-BinUtility adjustBinUtility(const BinUtility& bu, const RadialBounds& rBounds) {
+BinUtility adjustBinUtility(const BinUtility& bu, const RadialBounds& rBounds,
+                            const Transform3D& transform) {
   // Default constructor
   BinUtility uBinUtil;
+  if (transform.matrix() != Transform3D::Identity().matrix()) {
+    uBinUtil = BinUtility(std::make_shared<const Transform3D>(transform));
+  }
```
Comment (asalzburger) [inline Core/include/Acts/Utilities/BinAdjustment.hpp:36]: One could just test
`transform == Transform3D::Identity()` or even isApprox(), right?

**Comment by asalzburger** [comment]
@Corentin-Allaire  Can you add a test if the transform() is actually taken?

**Comment by Corentin-Allaire** [comment]
> @Corentin-Allaire Can you add a test if the transform() is actually taken?

I am not sure I understand what you mean by taken in this context ?

**Comment by asalzburger** [comment]
> > @Corentin-Allaire Can you add a test if the transform() is actually taken?
> 
> I am not sure I understand what you mean by taken in this context ?

You make BinUtility without transform, access the bins, then manually transform the values, create a BinUtility with transform and recreate ...

**Comment by Corentin-Allaire** [comment]
> > > @Corentin-Allaire Can you add a test if the transform() is actually taken?
> > 
> > 
> > I am not sure I understand what you mean by taken in this context ?
> 
> You make BinUtility without transform, access the bins, then manually transform the values, create a BinUtility with transform and recreate ...

Ah ok, I understand. Should be easy to do !

File: Core/include/Acts/Utilities/BinAdjustment.hpp (hunk 1 above)
Comment (Corentin-Allaire) [reply Core/include/Acts/Utilities/BinAdjustment.hpp:36]: I replace the != with isApprox

**Comment by Corentin-Allaire** [comment]
@asalzburger If you have no other issue/question do you think you could merge this in ? This way I could run a nightly during the night and test this in Athena tomorrow.

**Pull request merged**